File: atividade-4/030_uniao/030_uniao.js
//Faça uma função que receba duas listas a e b
//e retorne uma lista das chaves de a e b sem repetição

const list = [2, 5, 4, 9, 6, 10, 11, 12];
const list2 = [1, 2, 3, 4, 7, 42, 31, 20, 6];

const uniao = (xs, ys) => xs.concat(ys.filter((y) => !xs.includes(y)));

console.log(uniao(list, list2));

File: atividade-3/modulo-4/030_uniao/030_uniao.hs
module Main where

-- Faça uma função que receba duas listas a e b
-- e retorne uma lista das chaves de a e b sem repetição

uniao :: [Int] -> [Int] -> [Int]
uniao xs ys = xs ++ [y | y <- ys, not (elem y xs)]

main = do
  a <- readLn :: IO [Int]
  b <- readLn :: IO [Int]
  print $ uniao a b